tests/test_telemetry.py

```diff
@@ -597,6 +598,88 @@ def test_app_started_product_info(self):
                     "appsec" in products
                 ), "Product information is not accurately reported by telemetry on app-started event"
 
+    @bug(library="java", reason="APMAPI-969")
+    def test_config_telemetry_completeness(self):
+        """
+        Assert that config telemetry is handled properly by telemetry intake
+
+        ⚠️ Did this test just fail? Read here! ⚠️
+        Some files are manually copied from dd-go from/to the following paths using tests/telemetry_intake/update.sh
+        from: https://github.com/DataDog/dd-go/blob/prod/trace/apps/tracer-telemetry-intake/telemetry-payload/static/
+        to: tests/telemetry_intake/static
+
+        If this test fails, it means that a telemetry key was found in config telemetry that does not
+        exist in any of the files listed above in dd-go
+        The impact is that telemetry will not be reported to the Datadog backend won't be unusable
+
+        To fix this, you must update dd-go to either
+        1) Add an exact config key to match config_norm_rules.json
+        2) Add a prefix that matches the config keys to config_prefix_block_list.json
+        3) Add a prefix rule that fits an existing prefix to config_aggregation_list.json
+        4) (Discouraged) Add a language-specific rule to <lang>_config_rules.json
+
+        Once dd-go is updated, you can copy over the files to this repo and merge them in as part of your changes
+        This can be done by running the following from the src root
+
+        Usage: ./tests/telemetry_intake/update.sh
+        """
+
+        def load_telemetry_json(filename):
+            with open(f"tests/telemetry_intake/static/{filename}.json", encoding="utf-8") as fh:
+                return json.load(fh)
+
+        def lowercase_all(lst: list[str]):
+            return [x.lower() for x in lst]
+
+        config_norm_rules = load_telemetry_json("config_norm_rules")
+        config_prefix_block_list = load_telemetry_json("config_prefix_block_list")
+        config_aggregation_list = load_telemetry_json("config_aggregation_list")
+
+        lang_configs = {}
+        for lang in ["dotnet", "go", "jvm", "nodejs", "php", "python", "ruby"]:
+            lang_configs[lang] = load_telemetry_json(lang + "_config_rules")
+
+        for data in interfaces.library.get_telemetry_data(flatten_message_batches=True):
+            if not is_v2_payload(data):
+                continue
+            if get_request_type(data) == "app-started":
+                language_name = data["request"]["content"]["application"]["language_name"]
+
+                lang_config = lang_configs.get(language_name) or {}
+
+                allowed_config_keys = [*config_norm_rules] + [*(lang_config.get("normalization_rules") or {})]
+                allowed_config_values = [*config_norm_rules.values()] + [
```

Comment:
nit: We don't need to unpack the values. I think using list(...) is a bit cleaner here.

```suggestion
                allowed_config_values = list(config_norm_rules.values())+ [
```

Reply:
Added